File: python/loccount/linesofcode.py
def create_locstat(filename, lines):
    total = len(lines)
    loc = len([line for line in lines if is_loc(line)])
    result = (filename, total, loc)
    return result


def is_loc(line):
    return (line.lstrip() != "") & (not line.lstrip().startswith("#"))

File: csharp/loccount/loccount/loccount/LinesOfCode.cs
using System.Collections.Generic;
using System.Linq;

namespace loccount
{
    public static class LinesOfCode
    {
        public static LOCstat CountLines(IEnumerable<string> lines, string filename) {
            var locStat = new LOCstat();
            locStat.Filename = filename;
            locStat.Total = lines.Count();
            locStat.Loc = lines.Count(IsCodeLine);
            return locStat;
        }

        private static bool IsCodeLine(string line) {
            return !string.IsNullOrWhiteSpace(line) && !line.TrimStart().StartsWith("//");
        }
    }
}